EmployeeLib/Employee.cs

```diff
@@ -1,24 +1,34 @@
 ﻿namespace EmployeeLib
 {
-    public class Employee
+    public abstract class EmployeeBase
     {
         public string FirstName { get; set; }
         public string LastName { get; set; }
-        public Employee Manager { get; set; } = null;
         public decimal Salary { get; set; }
-
-        public virtual void AssignManager(Employee manager)
-        {
-            // Simulate doing other tasks here - otherwise, this should be
-            // a property set statement, not a method.
-            Manager = manager;
-        }
-
         public virtual void CalculatePerHourRate(int rank)
         {
             decimal baseAmount = 12.50M;
 
             Salary = baseAmount + (rank * 2);
         }
     }
+
+    public interface ISupervised<T>
```

Comment:
I think IHaveManager is better name for this, you can leave it as it is, but we need to uniform the code.
having a manager and a supervisor which it's basically the same is a bit hard to follow